casadm/cas_lib.c

```diff
@@ -1904,12 +1904,39 @@ int remove_core(unsigned int cache_id, unsigned int core_id,
 	if (run_ioctl_interruptible(fd, KCAS_IOCTL_REMOVE_CORE, &cmd,
 			"Removing core", cache_id, core_id) < 0) {
 		close(fd);
-		if (OCF_ERR_FLUSHING_INTERRUPTED == cmd.ext_err_code) {
-			cas_printf(LOG_ERR, "You have interrupted removal of core. CAS continues to operate normally.\n");
+		if (cmd.ext_err_code == OCF_ERR_FLUSHING_INTERRUPTED) {
+			if (detach) {
+				cas_printf(LOG_ERR, "You have interrupted "
+						"detaching of core. CAS "
+						"continues to operate "
+						"normally.\n");
+			} else {
+				cas_printf(LOG_ERR, "You have interrupted "
+						"removal of core. CAS "
+						"continues to operate "
+						"normally.\n");
+			}
```

Comment:
You could deduplicate this by writing `cas_printf(LOG_ERR, "You have... %s ... \n", "detaching" ? detach : "removal")`

Reply:
Thanks! It would be more aesthetic.

Reply:
Thanks! It would be more aesthetic.